internal/clients/http/image_builder/image_client.go

```diff
@@ -123,8 +140,35 @@ func (c *ibClient) fetchImageStatus(ctx context.Context, composeID string) (*Upl
 	}
 
 	if resp.JSON200.ImageStatus.Status != ImageStatusStatusSuccess {
-		logger.Warn().Msg("Image status in not ready")
+		logger.Warn().Msg("Compose status is not ready")
 		return nil, http.ImageStatusErr
 	}
+
 	return resp.JSON200.ImageStatus.UploadStatus, nil
 }
+
+func (c *ibClient) checkClone(ctx context.Context, composeID string) (*UploadStatus, error) {
+	logger := logger(ctx)
+	logger.Trace().Msgf("Fetching image status %v from clones", composeID)
+
+	resp, err := c.client.GetCloneStatusWithResponse(ctx, composeID, headers.AddImageBuilderIdentityHeader)
+	if err != nil {
+		logger.Warn().Err(err).Msg("Failed to fetch image status from image builder")
+		return nil, fmt.Errorf("cannot get compose status: %w", err)
+	}
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			return nil, fmt.Errorf("fetch image status call: %w", http.ComposeNotFoundErr)
```

Comment:
This is `checkClone` so:

```suggestion
			return nil, fmt.Errorf("fetch image status call: %w", http.CloneNotFoundErr)
```